### PR TITLE
Coll.updated implementation

### LINKED ISSUE #314: [150 SigmaUSD] Implement `Coll.indices` method
- Add `SMethod` to `ergotree-ir::types::scoll`(method id is 29); 
- Add `EvalFn` in `ertotree-interpreter::eval::scoll` and include it in `ergotree-interpreter::eval::smethod_eval_fn`

See existing method implementations in `scoll` and `sbox` for examples;

Reference Scala version:
eval - https://github.com/ScorexFoundation/sigmastate-interpreter/blob/b60edae20f2af3352e4064d139d45a15fc507f7d/library-impl/src/main/scala/special/collection/CollsOverArrays.scala#L74

### LINKED ISSUE #358: [150 SigmaUSD] Implement `Coll.updated` method
- Add `SMethod` to `ergotree-ir::types::scoll`(method id is 20); 
- Add `EvalFn` in `ertotree-interpreter::eval::scoll` and include it in `ergotree-interpreter::eval::smethod_eval_fn`

See existing method implementations in `scoll` and `sbox` for examples;

Reference Scala version:
https://github.com/ScorexFoundation/sigmastate-interpreter/blob/0c5c5b515ccb452536e440469b56d763d5ba30ec/sigmastate/src/main/scala/sigmastate/types.scala#L1600-L1613

### PR DESCRIPTION
**Pull request by ApexTheory**
PR for https://github.com/ergoplatform/sigma-rust/issues/358

FYI this branch is based off the [Coll.indices PR branch](https://github.com/ergoplatform/sigma-rust/pull/363) to avoid any merge conflicts.  

EDIT(@greenhat): Close #358; Close #314;

### CODE REVIEW

File: ergotree-interpreter/src/eval/scoll.rs [hunk 1]
```diff
@@ -352,4 +409,67 @@ mod tests {
...
+            vec![1i32.into(), 5i64.into()],
+        )
+        .unwrap()
+        .into();
+        let res = eval_out_wo_ctx::<Vec<i64>>(&expr);
+        assert_eq!(res, vec![1i64, 5i64, 3i64]);
+    }
+
+    #[test]
+    fn eval_update_oob() {
```
Comment (greenhat) [inline ergotree-interpreter/src/eval/scoll.rs:462]: Nice!

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking great! 
Could you implement my comments from #363 here? So we can drop #363 and merge only this one. Next time you can just create one PR. There is no strict rule 1 PR = 1 issue. :)

**Comment by ApexTheory** [comment]
@greenhat I incorporated your suggestions from Coll.indices here.  This PR has the solution for update & indices now.

File: ergotree-ir/src/types/scoll.rs [hunk 2]
```diff
@@ -18,7 +18,11 @@ pub const INDEX_OF_METHOD_ID: MethodId = MethodId(26);
 /// Coll.flatmap
 pub const FLATMAP_METHOD_ID: MethodId = MethodId(15);
 /// Coll.zip
-pub const ZIP_METHOD_ID: MethodId = MethodId(14);
+pub const ZIP_METHOD_ID: MethodId = MethodId(29);
```
Comment (greenhat) [inline ergotree-ir/src/types/scoll.rs:21]: Good catch! I appreciate it! Thank you!

**Review by greenhat — APPROVED** [review]
Looking great! Thank you!

**Pull request merged**